src/hardware/pounder/attenuators.rs

```diff
@@ -68,6 +67,9 @@ pub trait AttenuatorInterface {
         // care) would contain erroneous data.
         let attenuation_code = (!channels[channel as usize]) >> 2;
 
+        // The write-back transfer is performed. Staging register is now restored.
+        self.transfer_attenuators(&mut channels)?;
```

Comment:
The first transfer reads back the attenuation value and replace it with 0s (31.5 dB). When you trigger the second transfer immediately, you read back the the 0s (31.5 dB) and restore the original attenuation before we performed the read.

The original code does the read back-to-back, so you will always read 31.5 dB no matter what, because it is the dummy value that we set a few lines before this.

I only used the value to check the equality of attenuation before overwriting it, so all it does is do prevent unnecessary overwriting.

Reply:
I think the better approach is to copy `channels` into a new buffer for the write-back. That way, we keep around the `channels` value (which mimics what's actually in the staging register). Keeping them `channels` valid for only 1 line of code seems like a quick foot-gun. for the future.